python/task_manager/background_task_manager.py

```diff
@@ -0,0 +1,558 @@
+# Copyright (c) 2015 Shotgun Software Inc.
+# 
+# CONFIDENTIAL AND PROPRIETARY
+# 
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import traceback
+
+import sgtk
+from sgtk.platform.qt import QtCore
+from sgtk import TankError
+
+from .background_task import BackgroundTask
+from .worker_thread import WorkerThread
+
+class BackgroundTaskManager(QtCore.QObject):
+    """
+    Main task manager class. Manages a queue of tasks running them asynchronously through
+    a pool of worker threads.
+
+    Note that the BackgroundTaskManager class itself is reentrant but not thread-safe so its methods should only 
+    be called from the thread it is created in. Typically this would be the main thread of the application.
+    
+    
+    :signal task_completed(uid, group, result): Emitted when a task has been completed.
+        The ``uid`` parameter holds the unique id associated with the task, 
+        the ``group`` is the group that the task is associated with and 
+        the ``result`` is the data returned by the task. 
```

Comment:
Minor: Maybe put these signal args on lines of their own? I always prefer that to quick view over paragraph form? Just a thought.